server/cluster/cluster.go

```diff
@@ -390,24 +397,84 @@ func (c *RaftCluster) checkServices() {
 	}
 }
 
+// checkTSOService checks the TSO service.
+func (c *RaftCluster) checkTSOService() {
```

Comment:
can you add more comments for this function and inside this function. Seems there are too many situations in this function